SharpRepository.Tests.Integration/RepositoryAddTests.cs

```diff
@@ -19,7 +19,16 @@ public void Add_Should_Save_And_Assigned_New_Id(IRepository<Contact, string> rep
             repository.Add(contact);
             contact.ContactId.ShouldNotBeEmpty();
         }
-        
+
+        [ExecuteForAllRepositoriesExcept(RepositoryType.RavenDb, Reason = "Depends on driver to generate a value")]
```

Comment:
I added a new commit with a unit test for RavenDb to test that it throws `NotSupportedException` if `GenereateKeyOnAdd` is set to `false`. I think it's the best we can do if `GenerateKeyOnAdd` should be defined in `RepositoryBase`.